### PR TITLE
UI: Replace buttons for HDS in component templates part 2

### PR DESCRIPTION
**Pull request by hellobontempo**
Some of the `<LinkTo>` components have `class="button"` and therefore look like buttons but are technically not. These also need to be swapped for `<Hds::Button>` when used in places like forms, so this PR tackles those missed during the initial button replacement sweep through these files.

Follow on to https://github.com/hashicorp/vault/pull/23698

### CODE REVIEW

File: ui/app/templates/components/clients/config.hbs [hunk 1]
```diff
@@ -36,12 +36,10 @@
       {{/each}}
     </div>
     <div class="field is-grouped-split box is-fullwidth is-bottomless">
-      <div class="control">
```
Comment (hellobontempo) [inline ui/app/templates/components/clients/config.hbs:39]: Screenshot 2023-11-03 at 3 44 56 PM

File: ui/app/templates/components/generated-item.hbs
Comment (hellobontempo) [inline ui/app/templates/components/generated-item.hbs:1]: Screenshot 2023-11-03 at 3 50 17 PM

File: ui/app/templates/components/secret-create-or-update.hbs [hunk 2]
```diff
@@ -116,14 +116,19 @@
     {{/if}}
 
     <div class="field is-grouped box is-fullwidth is-bottomless">
-      <div class="control">
+      <Hds::ButtonSet>
         <Hds::Button @text="Save" type="submit" disabled={{@buttonDisabled}} data-test-secret-save={{true}} />
-      </div>
-      <div class="control">
-        <SecretLink @mode="list" @secret={{@model.parentKey}} class="button">
```
Comment (hellobontempo) [inline ui/app/templates/components/secret-create-or-update.hbs:123]: Another example of replacement https://github.com/hashicorp/vault/pull/23698/files#diff-5a1b6fbb9e7b80234fab1843f373c27c90f123c21dbf303efaa765da9563ef96L95-L101

File: ui/app/templates/components/secret-create-or-update.hbs [hunk 3]
```diff
@@ -116,14 +116,19 @@
...
-          Cancel
-        </SecretLink>
-      </div>
+        {{#if @model.path}}
+          <Hds::Button
+            @text="Cancel"
+            @color="secondary"
+            @route="vault.cluster.secrets.backend.list"
+            @model={{@model.path}}
```
Comment (hellobontempo) [inline ui/app/templates/components/secret-create-or-update.hbs:126]: Purposefully changed to `path` from `parentKey` so it's brought back to the right list view (previously it was going too far up the tree

Comment (hellobontempo) [reply ui/app/templates/components/secret-create-or-update.hbs:126]: ## fix 

> on `Cancel` form should go back to the `one/two` list view

fix

## bug

> previously was going back two levels, notice on `Cancel` we go back to `one/` even though we were creating a new secret from `one/two/`

bug

**Review by kiannaquach — APPROVED** [review]
appreciate you for tackling this on a Friday! 🎉

**Review by hashishaw — APPROVED** [review]
Thanks!

**Pull request merged**